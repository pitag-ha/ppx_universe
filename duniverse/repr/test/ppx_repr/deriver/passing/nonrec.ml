type t = unit [@@deriving repr]
type t_alias = unit [@@deriving repr]

(* Ensure that 'nonrec' assertions are respected *)
module S1 : sig
  type nonrec t = t list [@@deriving repr]
  type nonrec t_alias = t_alias list [@@deriving repr]
end = struct
  type nonrec t = t list [@@deriving repr]
  type nonrec t_alias = t_alias list [@@deriving repr]
end

(* Now test the interaction of 'nonrec' with custom naming *)
module S2 : sig
  type nonrec t = t list [@@deriving repr { name = "t_repr" }]
  type nonrec t_alias = t_alias list [@@deriving repr { name = "t_repr" }]
end = struct
  type nonrec t = t list [@@deriving repr { name = "t_repr" }]
  type nonrec t_alias = t_alias list [@@deriving repr { name = "t_repr" }]
end
