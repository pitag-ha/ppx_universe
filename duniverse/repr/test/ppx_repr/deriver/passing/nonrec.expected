type t = unit[@@deriving repr]
include struct let t = Repr.unit end[@@ocaml.doc "@inline"][@@merlin.hide ]
type t_alias = unit[@@deriving repr]
include struct let t_alias_t = Repr.unit end[@@ocaml.doc "@inline"][@@merlin.hide
                                                                    ]
module S1 :
  sig
    type nonrec t = t list[@@deriving repr]
    include sig val t : t Repr.t end[@@ocaml.doc "@inline"][@@merlin.hide ]
    type nonrec t_alias = t_alias list[@@deriving repr]
    include sig val t_alias_t : t_alias Repr.t end[@@ocaml.doc "@inline"]
    [@@merlin.hide ]
  end =
  struct
    type nonrec t = t list[@@deriving repr]
    include struct let t = Repr.list t end[@@ocaml.doc "@inline"][@@merlin.hide
                                                                   ]
    type nonrec t_alias = t_alias list[@@deriving repr]
    include struct let t_alias_t = Repr.list t_alias_t end[@@ocaml.doc
                                                            "@inline"]
    [@@merlin.hide ]
  end 
module S2 :
  sig
    type nonrec t = t list[@@deriving repr { name = "t_repr" }]
    include sig val t_repr : t Repr.t end[@@ocaml.doc "@inline"][@@merlin.hide
                                                                  ]
    type nonrec t_alias = t_alias list[@@deriving repr { name = "t_repr" }]
    include sig val t_repr : t_alias Repr.t end[@@ocaml.doc "@inline"]
    [@@merlin.hide ]
  end =
  struct
    type nonrec t = t list[@@deriving repr { name = "t_repr" }]
    include struct let t_repr = Repr.list t end[@@ocaml.doc "@inline"]
    [@@merlin.hide ]
    type nonrec t_alias = t_alias list[@@deriving repr { name = "t_repr" }]
    include struct let t_repr = Repr.list t_alias_t end[@@ocaml.doc
                                                         "@inline"][@@merlin.hide
                                                                    ]
  end 
